code/processes/tickerlogreplay.q

```diff
@@ -197,20 +208,21 @@ replaylog:{[logfile]
          @[`.;`upd;:;.replay.initialupd]];
 	@[`.;`upd;:;.replay.realupd]];
  .replay.tablecounts:.replay.errorcounts:.replay.pathlist:()!();
- .replay.replaydate:"D"$-10#string logfile;
+ // extract date from timestamp if stp, extract date from date if tp
+ .replay.replaydate:"D"$$[segmentedmode;-6_-14#;-10#]string logfile;
  if[ .replay.clean and (`$st:string .replay.replaydate) in key hdbdir;
     .lg.o[`replay;"HDB directory already contains ",st," partition. Deleting from the HDB directory"];
     .os.deldir .os.pth[.Q.par[hdbdir;.replay.replaydate;`]]; // delete the current dates HDB directory before performing replay
   ];
  if[lastmessage<firstmessage; .lg.o[`replay;"lastmessage (",(string lastmessage),") is less than firstmessage (",(string firstmessage),"). Not replaying log file"]; :()];
  .lg.o[`replay;"replaying data from logfile ",(string logfile)," from message ",(string firstmessage)," to ",(string lastmessage),". Message indices are from 0 and inclusive - so both the first and last message will be replayed"];
  // when we do the replay, need to move the indexing, otherwise we wont replay the last message correctly
- -11!($[lastmessage<0W; lastmessage+1;lastmessage];logfile);
+ $[`stpmeta~`$-7#string logfile;.lg.o[`replay;"skipping meta table: ",string[logfile]];-11!($[lastmessage<0W; lastmessage+1;lastmessage];logfile)];
```

Comment:
Can you break this case statement onto a couple of lines 
Also if it's the stpmeta file, would be better to jump out at this point
:()
rather than printing the log messages that won't make sense

Reply:
This is now changed